File: excellent-educators-web/lib/core/network/api_exception.dart
class ApiException implements Exception {
  ApiException({
    required this.message,
    this.code,
    this.statusCode,
    this.details,
  });

  final String message;
  final String? code;
  final int? statusCode;
  final Object? details;

  @override
  String toString() => message;
}

File: excellent-educators-api/app/Exceptions/ApiException.php
<?php

namespace App\Exceptions;

use Symfony\Component\HttpKernel\Exception\HttpException;

class ApiException extends HttpException
{
    public function __construct(
        public readonly string $errorCode,
        string $message,
        int $status = 400,
        public readonly mixed $details = null,
    ) {
        parent::__construct($status, $message);
    }
}
